plonk/src/proof_system/snark.rs

```diff
@@ -76,7 +76,7 @@ where
         let domain_size = circuit.eval_domain_size()?;
         let num_inputs = circuit.num_inputs();
         let supported_hiding_bound = 2;
-        if srs.0.max_degree() < domain_size + supported_hiding_bound {
+        if srs.0.max_degree() < circuit.srs_size()? {
```

Comment:
In line 99:
`let (commit_key, open_key) = trim(&srs.0, domain_size + supported_hiding_bound);`

Maybe we can define variable `srs_size = circuit.srs_size()` first, and then replace `domain_size + supported_hiding_bound` with `srs_size`, then we don't need to define `supported_hiding_bound` anymore.